packages/squiggle-lang/src/fr/units.ts

```diff
@@ -19,15 +21,22 @@ const makeUnitFn = (
     examples: [`3${shortName} // ${3 * multiplier}`],
     isUnit: true,
     definitions: [
-      makeDefinition([frNumber], frNumber, ([x]) => x * multiplier),
+      format
+        ? makeDefinition([frNumber], frForceBoxed(frNumber), ([x]) => {
+            return {
+              value: x * multiplier,
+              args: new BoxedArgs({ numberFormat: format }),
+            };
+          })
+        : makeDefinition([frNumber], frNumber, ([x]) => x * multiplier),
     ],
   });
 };
 
 export const library = [
   makeUnitFn("n", "nano", 1e-9),
   makeUnitFn("m", "mili", 1e-3),
-  makeUnitFn("%", "percent", 1e-2),
+  makeUnitFn("%", "percent", 1e-2, ".2%"),
```

Comment:
`.2~p` seems slightly better; it rounds larger numbers too much but doesn't lose heading significant digits in small numbers:

```
> format('.2~p')(3456789)
'350000000%'
> format('.2~p')(34.56789)
'3500%'
> format('.2~p')(0.3456789)
'35%'
> format('.2~p')(0.003456789)
'0.35%'
> format('.2~p')(0.00003456789)
'0.0035%'
```

Compared to `.2%`:
```
> format('.2%')(3456789)
'345678900.00%'
> format('.2%')(34.56789)
'3456.79%'
> format('.2%')(0.3456789)
'34.57%'
> format('.2%')(0.003456789)
'0.35%'
> format('.2%')(0.00003456789)
'0.00%'
```

Reply:
Ah, that's neat, happy to change.